util/tippingContractUtil.js

```diff
@@ -28,11 +28,47 @@ tippingContractUtil.getTipsRetips = (state) => {
     return data;
   });
 
-  const tips = state.tips.map(([id, data]) => {
+  const tips = state.tips.map(([id, tipTypeData]) => {
+    const [tipType, tipData] = Object.entries(tipTypeData)[0];
+    switch (tipType) {
+      case 'AeTip':
+        data = tipData[0];
+        data.url_id = tipData[1];
+        data.amount = tipData[2];
+        data.claim_gen = tipData[3];
+        break;
+      case 'TokenTip':
+        data = tipData[0];
+        data.url_id = tipData[1];
+        data.token = tipData[2].token;
+        data.token_amount = tipData[2].amount;
+        data.claim_gen = tipData[3];
+        data.amount = 0;
+        break;
+      case 'DirectAeTip':
+        data = tipData[0];
+        data.receiver = tipData[1];
+        data.amount = tipData[2];
+        break;
+      case 'DirectTokenTip':
+        data = tipData[0];
+        data.receiver = tipData[1];
+        data.token = tipData[2].token;
+        data.token_amount = tipData[2].amount;
+        data.amount = 0;
+        break;
+      default:
+        data = tipTypeData; // Fallback for old contract state format
+        break;
+    }
+
+    data.type = data.type ? data.type : 'AeTip';
     data.id = id;
-    data.url = findUrl(data.url_id);
-    data.retips = findRetips(id, data.url_id);
-    data.claim = findClaimGen(data.claim_gen, data.url_id);
+
+    const hasClaim = data.claim_gen !== undefined && data.url_id !== undefined;
```

Comment:
can we rename this to something like isDirectTip?

Reply:
technically it decides if it has a claim, also this is just used internally to the util and is not exposed, so I think its fine as is